src/tls/certificate.rs

```diff
@@ -218,32 +240,57 @@ fn parse_key(mut key: &[u8]) -> Result<PrivateKeyDer<'static>, Error> {
 impl WorkloadCertificate {
     pub fn new(key: &[u8], cert: &[u8], chain: Vec<&[u8]>) -> Result<WorkloadCertificate, Error> {
         let cert = parse_cert(cert.to_vec())?;
-        let chain = chain
-            .into_iter()
+
+        // The Istio API does something pretty unhelpful, by providing a single chain of certs.
+        // The last one is the root. However, there may be multiple roots concatenated in that last cert,
+        // so we will need to split them.
+        let Some(raw_root) = chain.last() else {
+            return Err(Error::InvalidRootCert(
+                "no root certificate present".to_string(),
+            ));
+        };
+        let roots = parse_cert_multi(raw_root)?;
+        let chain = chain[..cmp::max(0, chain.len() - 1)]
+            .iter()
             .map(|x| x.to_vec())
             .map(parse_cert)
             .collect::<Result<Vec<_>, _>>()?;
         let key: PrivateKeyDer = parse_key(key)?;
 
-        let mut roots = RootCertStore::empty();
-        roots.add_parsable_certificates(chain.iter().last().map(|c| c.der.clone()));
+        let mut roots_store = RootCertStore::empty();
+        roots_store.add_parsable_certificates(roots.iter().map(|c| c.der.clone()));
```

Comment:
Should we check the return status of `add_parsable_certificates`? Feels like we should at least log an error if we find unparseable ones.

https://docs.rs/rustls/latest/rustls/struct.RootCertStore.html#method.add_parsable_certificates
> Returns the number of certificates added, and the number that were ignored.